vmdb/lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -88,9 +88,15 @@ def self.expose(*args)
         next if method_name.to_sym == :id
         self.association = method_name if options[:association]
         define_method(method_name) do |*params|
-          ret = object_send(options[:method] || method_name, *params)
+          method = options[:method] || method_name
+          ret = object_send(method, *params)
           return options[:override_return] if options.has_key?(:override_return)
-          return wrap_results(ret)
+          reflection = @object.class.reflection_with_virtual(method)
+          if reflection && reflection.collection?
```

Comment:
Are you sure we should do that *in* wrap_results?  It means we'd have to pass the reflection in to the method too.  Currently `wrap_results` doesn't know what a reflection is (and I don't really want to teach it).